deploy/src/erc_to_native/home.js

```diff
@@ -0,0 +1,201 @@
+    data: upgradeToBridgeVHomeData,
+    nonce: homeNonce,
+    to: storageValidatorsHome.options.address,
+    privateKey: deploymentPrivateKey,
+    url: HOME_RPC_URL
+  })
+  assert.equal(txUpgradeToBridgeVHome.status, '0x1', 'Transaction Failed')
+  homeNonce++
+
+  console.log('\ninitializing Home Bridge Validators with following parameters:\n')
+  console.log(
+    `REQUIRED_NUMBER_OF_VALIDATORS: ${REQUIRED_NUMBER_OF_VALIDATORS}, VALIDATORS: ${VALIDATORS}`
+  )
+  bridgeValidatorsHome.options.address = storageValidatorsHome.options.address
+  const initializeData = await bridgeValidatorsHome.methods
+    .initialize(REQUIRED_NUMBER_OF_VALIDATORS, VALIDATORS, HOME_OWNER_MULTISIG)
+    .encodeABI({ from: DEPLOYMENT_ACCOUNT_ADDRESS })
+  const txInitialize = await sendRawTxHome({
+    data: initializeData,
+    nonce: homeNonce,
+    to: bridgeValidatorsHome.options.address,
+    privateKey: deploymentPrivateKey,
+    url: HOME_RPC_URL
+  })
+  assert.equal(txInitialize.status, '0x1', 'Transaction Failed')
+  homeNonce++
+
+  console.log('transferring proxy ownership to multisig for Validators Proxy contract')
+  const proxyDataTransfer = await storageValidatorsHome.methods
+    .transferProxyOwnership(HOME_UPGRADEABLE_ADMIN_VALIDATORS)
+    .encodeABI()
+  const txProxyDataTransfer = await sendRawTxHome({
+    data: proxyDataTransfer,
+    nonce: homeNonce,
+    to: storageValidatorsHome.options.address,
+    privateKey: deploymentPrivateKey,
+    url: HOME_RPC_URL
+  })
+  assert.equal(txProxyDataTransfer.status, '0x1', 'Transaction Failed')
+  homeNonce++
+
+  console.log('\ndeploying homeBridge storage\n')
+  const homeBridgeStorage = await deployContract(EternalStorageProxy, [], {
+    from: DEPLOYMENT_ACCOUNT_ADDRESS,
+    nonce: homeNonce
+  })
+  homeNonce++
+  console.log('[Home] HomeBridge Storage: ', homeBridgeStorage.options.address)
+
+  console.log('\ndeploying homeBridge implementation\n')
+  const homeBridgeImplementation = await deployContract(HomeBridge, [], {
+    from: DEPLOYMENT_ACCOUNT_ADDRESS,
+    nonce: homeNonce
+  })
+  homeNonce++
+  console.log('[Home] HomeBridge Implementation: ', homeBridgeImplementation.options.address)
+
+  console.log('\nhooking up HomeBridge storage to HomeBridge implementation')
+  const upgradeToHomeBridgeData = await homeBridgeStorage.methods
+    .upgradeTo('1', homeBridgeImplementation.options.address)
+    .encodeABI({ from: DEPLOYMENT_ACCOUNT_ADDRESS })
+  const txUpgradeToHomeBridge = await sendRawTxHome({
+    data: upgradeToHomeBridgeData,
+    nonce: homeNonce,
+    to: homeBridgeStorage.options.address,
+    privateKey: deploymentPrivateKey,
+    url: HOME_RPC_URL
+  })
+  assert.equal(txUpgradeToHomeBridge.status, '0x1', 'Transaction Failed')
+  homeNonce++
+
+  let blockRewardAddress = null
+  if (BLOCK_REWARD_ADDRESS) {
+    blockRewardAddress = BLOCK_REWARD_ADDRESS
+  } else {
+    console.log('\n[Home] deploying Block Reward contract (dev only)')
```

Comment:
If the deploy is made with `NODE_ENV=production`, `envalid` will throw. Sorry that isn't clear here.

But if you think it's safer to just throw, we can separate the mock block reward deploy to another script, sure.

Reply:
Usage of `NODE_ENV` is not evident for me at least. 
Sometimes, it will be useful to deploy some contracts from this repo separately. For example, I added ability to deploy an ERC20 token in #87 (https://github.com/poanetwork/poa-bridge-contracts/pull/87/commits/7734f579fb62dc1e674205656179bd37ac7eb097).

Reply:
That makes sense, and it may be helpful for the e2e tests too. I'll make a `deployBlockReward` script,